projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -346,7 +346,7 @@ public static function parse( $attributes, $content ) {
 			} else {
 				// Submit form to the post permalink
 				$url = get_permalink();
-				if ( $page ) {
+				if ( $multipage && $page ) {
```

Comment:
Fix is to only add the `?page` query param to the form submission url if the post has more than one page. The `$multipage` global stores whether it's a multi page post with a value of `0` or `1` so we can use that to check.

Perhaps in classic themes it was possible for users to navigate to a URL like that and not end up on a 404 page, but it doesn't seem to be the case in block themes.